### PR TITLE
PHPUnit 7

### PR DESCRIPTION
**Pull request by carusogabriel**
WIP as [Symfony PHPUnit Bridge will need some compatibility](https://travis-ci.org/doctrine/dbal/jobs/336479070#L545).

### CODE REVIEW

**Comment by Majkl578** [comment]
ref symfony/symfony#26017 / symfony/symfony#26024

**Comment by Majkl578** [comment]
Blocking issues/PRs were closed/merged, anything else blocking now?

**Comment by carusogabriel** [comment]
@Majkl578 As mentioned in https://github.com/symfony/symfony/pull/26139, Symfony still not compatible with PHPUnit 7. I have already used `dev-master` to test, but nothing for the moment. I'll rerun tests, but if continuous to fail, I'll close it and wait for them.

**Comment by morozov** [comment]
Can we just bump PHPUnit requirements on our end and update the bridge later? I see that we run PHPUnit directly and also have a dependency on it directly. What's the point of using the bridge?

**Comment by Majkl578** [comment]
> What's the point of using the bridge?

Depecation notices I suppose, added in #2846.

**Comment by Majkl578** [comment]
Well, the bridge itself is [part of Symfony](https://github.com/symfony/symfony/tree/master/src/Symfony/Bridge/PhpUnit) so I guess it meant _Symfony is not compatible, therefore the bridge is not compatible either_.

**Comment by morozov** [comment]
When You Marry ‘The One,’ You Also Marry The Family… eh.

**Comment by Majkl578** [comment]
Okay, I pushed a fix to the suite & bumped requirement directly into @carusogabriel's branch (this PR). For now it depends on the bridge `4.0.5@dev` until Symfony releases version with proper PHPUnit 7 support.

**Comment by Majkl578** [comment]
Build is green except the lowest deps, those will be fixed once stable version is released. 😊

**Comment by carusogabriel** [comment]
@Majkl578 Oh thanks, I was banging my head against version `3.x`, while the solution was in `4.x` :tada:

**Comment by Majkl578** [comment]
The fixes were merged into 3.4 and up so `^3.4.5|^4.0.5` would probably work too, but why keep it legacy, right? 😉

File: composer.json [hunk 1]
```diff
@@ -18,10 +18,10 @@
     },
     "require-dev": {
         "doctrine/coding-standard": "~2.0.0",
-        "phpunit/phpunit": "^6.3",
+        "phpunit/phpunit": "^7.0",
         "phpunit/phpunit-mock-objects": "!=3.2.4,!=3.2.5",
         "symfony/console": "^2.0.5||^3.0",
-        "symfony/phpunit-bridge": "^3.3"
+        "symfony/phpunit-bridge": "^4.0.5@dev"
```
Comment (Ocramius) [inline composer.json]: Can't :ship: until this is an `@dev`

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Majkl578) [reply composer.json]: Yup, waiting for Symfony to release, will drop `@dev` after that.

**Comment by Majkl578** [comment]
Updated with the `^3.4.5|^4.0.5` constraint. Ready to land.

**Review by Majkl578 — APPROVED** [review]

**Review by Ocramius — APPROVED** [review]
:ship:

**Pull request merged**